lib/proxy_api/external_ipam.rb

```diff
@@ -0,0 +1,278 @@
+    #   Response if subnet error:
+    #     {"code": 404, "error": "The specified subnet does not exist in External IPAM."}
+    #   Response if missing required params:
+    #     {"code": 400, "error": ["A 'cidr' parameter for the subnet must be provided(e.g. 100.10.10.0/24)","Missing 'ip' parameter. An IPv4 address must be provided(e.g. 100.10.10.22)"]}
+    #   Response if can't connect to External IPAM server
+    #     {"code": 500, "error": "Unable to connect to External IPAM server"}
+    def add_ip_to_subnet(ip, subnet, group = "")
+      raise "subnet cannot be nil" if subnet.nil?
+      response = parse post({}, "/subnet/#{subnet}/#{ip}?group=#{URI.escape(group.to_s)}")
+      raise ::Foreman::Exception.new(N_(response['error'])) if response['error']
+      response['code'] == 200
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to add IP %s to the subnet %s in External IPAM."), ip, subnet)
+    end
+
+    # Get a list of groups from External IPAM. A group is analagous to a 'section' in phpIPAM, and
+    # is a logical grouping of subnets/ips.
+    #
+    # Params: None
+    #
+    # Returns: An array of groups on success, or a hash with a "message" key
+    #          containing error on failure.
+    #
+    # Responses from Proxy plugin:
+    #   Response if success:
+    #     {"code": 200, "data": [
+    #       {name":"Test Group","description": "A Test Group"},
+    #       {name":"Awesome Group","description": "A totally awesome Group"}
+    #     ]}
+    #   Response if no groups exist:
+    #     {"code": 200, "data": []}
+    #   Response if groups are not supported:
+    #     {"code": 501, "error": "Groups are not supported"}
+    #   Response if can't connect to External IPAM server
+    #     {"code": 500, "error": "Unable to connect to External IPAM server"}
+    def get_groups
+      parse get("/groups")
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to obtain groups from External IPAM."))
+    end
+
+    # Get a group from External IPAM. A group is analagous to a 'section' in phpIPAM, and
+    # is a logical grouping of subnets/ips.
+    #
+    # Params: group: The name of the External IPAM group
+    #
+    # Returns: An External IPAM group on success, or a hash with an "error" key on failure.
+    #
+    # Responses from Proxy plugin:
+    #   Response if success:
+    #     {"code": 200, "data": {"name":"Awesome Section", "description": "Awesome Section"}}
+    #   Response if group doesn't exist:
+    #     {"code": 404, "message": "Not found"}
+    #   Response if groups are not supported:
+    #     {"code": 501, "error": "Groups are not supported"}
+    #   Response if can't connect to External IPAM server
+    #     {"code": 500, "error": "Unable to connect to External IPAM server"}
+    def get_group(group)
+      raise "group must be provided" if group.blank?
+      parse get("/groups/#{URI.escape(group)}")
```

Comment:
Can you also add the raise when `error` is returned to align behaviour with  previous methods if possible?